nav2_collision_monitor/src/source_base.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_collision_monitor/source_base.hpp"
+#include "nav2_collision_monitor/kinematics.hpp"
+
+#include <cmath>
+#include <exception>
+
+#include "tf2/convert.h"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+
+#include "nav2_util/node_utils.hpp"
+
+namespace nav2_collision_monitor
+{
+
+SourceBase::SourceBase()
+: tf_buffer_(nullptr), source_type_(SOURCE_BASE), source_name_(""),
+  source_topic_(""), source_frame_id_(""), base_frame_id_(""),
+  transform_tolerance_(0.0), max_time_shift_(0.0)
+{
+}
+
+SourceBase::SourceBase(
+  const nav2_util::LifecycleNode::WeakPtr & node,
+  std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+  const std::string source_name,
+  const std::string base_frame_id,
+  const double transform_tolerance,
+  const double max_time_shift)
+: node_(node), tf_buffer_(tf_buffer), source_type_(SOURCE_BASE), source_name_(source_name),
+  source_topic_(""), source_frame_id_(""), base_frame_id_(base_frame_id),
+  transform_tolerance_(transform_tolerance), max_time_shift_(max_time_shift)
+{
+  auto node_sptr = node_.lock();
+  if (node_sptr) {
+    RCLCPP_INFO(node_sptr->get_logger(), "Creating SourceBase");
+  }
+}
+
+SourceBase::~SourceBase()
+{
+  auto node = node_.lock();
+  if (node) {
+    RCLCPP_INFO(node->get_logger(), "Destroying SourceBase");
+  }
+
+  tf_buffer_.reset();
+}
+
+bool SourceBase::getParameters()
+{
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  try {
```

Comment:
Can't throw, doesn't need to return true/false

Reply:
Removed try-catch blocks and `bool` return type for SourceBase: https://github.com/ros-planning/navigation2/pull/2982/files#diff-7e24f9dbbee0e03a22353c8e2625792fc824ced725395a85665629f232ce4706R54